dlib/pixel.h

```diff
@@ -125,6 +125,13 @@ namespace dlib
         unsigned char red;
         unsigned char green;
         unsigned char blue;
+
+        inline bool operator == (const rgb_pixel& that) const
+        {
+            return this->red   == that.red
+                && this->green == that.green
+                && this->blue  == that.blue;
+        }
```

Comment:
I'm not sure about this one.  It creates a non-uniformity in the pixel types as many others don't allow ==.  Is this used in the code in this PR?

Reply:
Yes, it is used in the handling of PASCAL VOC 2012 classes, which have RGB codes.

If you don't want it here, I can move it to be "local" to the segmentation example code?

Reply:
Na. I changed my mind. :)

Leave this here. It’s cool.

Reply:
Although you don’t need the inline.  And add operator equal for the other pixel types in this file while here so things are basically uniform.

Reply:
Fixed.